### PR TITLE
Multiple tweaks to document text and definitions. 

### PR DESCRIPTION
**Pull request by waynebeaton**
I've changed the definitions of project and release to mirror the definitions already present in the document.

I've removed the discussion regarding very specific naming rules around Milestones and Release Candidates in the Releases section and replaced it with a recommendation/example.

I've introduced pseudo-legal capitalization style to all defined terms in the main document.

### CODE REVIEW

File: eclipse_development_process.adoc [hunk 1]
```diff
@@ -310,182 +310,180 @@ digraph {
...
+_(Most of this section is paraphrased from the excellent http://www.apache.org/dev/Release.html[Apache Software Foundation Releases FAQ]. The Eclipse community has many of the same beliefs about Releases as does the Apache community and their words were already excellent. The Apache Software Foundation Releases FAQ is distributed under the http://www.apache.org/licenses/LICENSE-2.0[Apache License, Version 2.0].)_
 
-Releases are, by definition, anything that is distributed outside of the committers of a project. If consumers are being directed to download a build, then that build has been released (modulo the exceptions below). All projects and committers must obey the Eclipse Foundation requirements on approving any release.
+Releases are, by definition, anything that is distributed outside of the Committers of a Project. If consumers are being directed to download a build, then that build has been Released (modulo the exceptions below). All Projects and Committers must obey the Eclipse Foundation requirements on approving any Release.
```
Comment (guw) [inline eclipse_development_process.adoc]: `Released` is also capitalized. Is this intentional?

File: eclipse_development_process.adoc [hunk 2]
```diff
@@ -310,182 +310,180 @@ digraph {
...
 
 [#6_4_Releases]
 === 6.4 Releases
 
-Any project, with exception of permanent incubators, may make a release. A release may include the code from any subset of the project's descendants.
+Any Project, with exception of Permanent Incubators, may make a Release. A Release may include the code from any subset of the Project's descendants.
 
-_(Most of this section is paraphrased from the excellent http://www.apache.org/dev/release.html[Apache Software Foundation Releases FAQ]. The Eclipse community has many of the same beliefs about Releases as does the Apache community and their words were already excellent. The Apache Software Foundation Releases FAQ is distributed under the http://www.apache.org/licenses/LICENSE-2.0[Apache License, Version 2.0].)_
+_(Most of this section is paraphrased from the excellent http://www.apache.org/dev/Release.html[Apache Software Foundation Releases FAQ]. The Eclipse community has many of the same beliefs about Releases as does the Apache community and their words were already excellent. The Apache Software Foundation Releases FAQ is distributed under the http://www.apache.org/licenses/LICENSE-2.0[Apache License, Version 2.0].)_
```
Comment (guw) [inline eclipse_development_process.adoc]: The link has been capitalized. Was your search and replace too broad?

File: eclipse_development_process.adoc [hunk 3]
```diff
@@ -310,182 +310,180 @@ digraph {
...
+The Project Team must provide (either directly or indirectly) a link between the distribution form of Release artifacts and the corresponding source code.
 
-*Exception 1: nightly and integration builds* During the process of developing software and preparing a release, various nightly and integration builds are made available to the developer community for testing purposes. Do not include any links on the project website, blogs, wikis, etc. that might encourage non-early-adopters to download and use nightly builds, release candidates, or any other similar package (links aimed at early-adopters and the project's developers are both permitted and encouraged). The only people who are supposed to know about such packages are the people following the developer mailing list and thus are aware of the limitations of such builds.
+A Project can make official Releases for one calendar year following a successful link:#6_3_5_Progress_Review[Progress Review] or link:#6_3_3_Release_Review[Release Review]. The Project Leadership Chain may--at its discretion--require that the Project engage in additional Reviews (e.g. a progress or Release Review) prior to issuing a Release.
```
Comment (guw) [inline eclipse_development_process.adoc:448]: Is this a special formatting with the two dashes `--` and no space separation?

File: eclipse_development_process.adoc [hunk 4]
```diff
@@ -310,182 +310,180 @@ digraph {
...
 
-The {ipPolicyUrl}[Eclipse IP Policy] must be followed at all times. Prior to issuing any release, the Project Lead must assert that the IP Policy has been followed and all approvals have been received.
+The {ipPolicyUrl}[Eclipse IP Policy] must be followed at all times. Prior to issuing any Release, the Project Lead must assert that the IP Policy has been followed and all approvals have been received.
 
-A project can make official releases for one calendar year following a successful link:#6_3_5_Progress_Review[progress review] or link:#6_3_3_Release_Review[release review]. The project leadership chain may--at its discretion--require that the project engage in additional reviews (e.g. a progress or release review) prior to issuing a release.
+The Project Team must provide (either directly or indirectly) a link between the distribution form of Release artifacts and the corresponding source code.
```
Comment (guw) [inline eclipse_development_process.adoc:446]: I wonder if `(either directly or indirectly)` is necessary. Should we just omit it?

File: eclipse_development_process.adoc [hunk 5]
```diff
@@ -310,182 +310,180 @@ digraph {
...
+*Exception 1: nightly and Integration Builds* During the process of developing software and preparing a Release, various nightly and Integration Builds are made available to the Developer community for testing purposes. Do not include any links on the Project website, blogs, wikis, etc. that might encourage non-early-adopters to download and use Nightly Builds, Release Candidates, or any other similar package (links aimed at early-adopters and the Project's Developers are both permitted and encouraged). The only people who are supposed to know about such packages are the people following the Developer mailing list and thus are aware of the limitations of such builds.
```
Comment (guw) [inline eclipse_development_process.adoc]: Should `nightly` be capitalized too?

File: terms.adoc [hunk 6]
```diff
@@ -99,7 +97,7 @@ Proposal Phase ::
 The Proposal Phase is a Phase during which a Project Proposal is presented to the community and Membership at Large to solicit feedback.
 
 Release ::
-A Release is a specifically identifiable set of artifacts that are collectively assigned a durable tag in source control and distributed in a semantically versioned durable form of that collection intended by the developers of the project for third-party use.
+A Release is a collection of project artifacts intended for distribution beyond the project Developers.
```
Comment (guw) [inline terms.adoc]: @waynebeaton I'm not happy with the mix of content and capitalization changes in one single commit. This makes it hard to follow.

FWIW, the capitalization changes look too broad and sometimes odd in the document.

File: eclipse_development_process.adoc (hunk 1 above)
Comment (waynebeaton) [reply eclipse_development_process.adoc]: Not intentional. Nice catch.

File: eclipse_development_process.adoc (hunk 2 above)
Comment (waynebeaton) [reply eclipse_development_process.adoc]: Another nice catch.

File: eclipse_development_process.adoc (hunk 3 above)
Comment (waynebeaton) [reply eclipse_development_process.adoc:448]: That converts into an em-dash. It is grammatically correct to **not** put spaces around an em dash.

File: eclipse_development_process.adoc (hunk 5 above)
Comment (waynebeaton) [reply eclipse_development_process.adoc]: Yes. Nice catch.

File: terms.adoc (hunk 6 above)
Comment (waynebeaton) [reply terms.adoc]: Understood. This is less than optimal. I should have done it with two passes.

The mixture does look odd. The original document had this mixture, and I removed it in 2015. It turns out that it is problematic to not have the mixed case when grouping this document with others written with the pseudo legal capitalization (e.g. IP Policy and Bylaws).

I took a pass through the definitions to make sure that they are consistent.

I'll send a case-insensitive red-line to the AC mailing list.

File: eclipse_development_process.adoc (hunk 4 above)
Comment (waynebeaton) [reply eclipse_development_process.adoc:446]: I'm going to leave it for now. I would like to avoid folks thinking that direct links in compiled content are required.

**Review by guw — APPROVED** [review]

**Review by msohn — APPROVED** [review]

**Review by Emily-Jiang — APPROVED** [review]
+1
